packages/entity-server/src/routes/hierarchy/MultiEntityDescendantsRoute.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { Route } from '@tupaia/server-boilerplate';
+import { EntityType } from '../../models';
+import { formatEntitiesForResponse } from './format';
+import {
+  MultiEntityRequest,
+  MultiEntityRequestParams,
+  RequestBody,
+  MultiEntityRequestQuery,
+  EntityResponse,
+} from './types';
+
+export type MultiEntityDescendantsRequest = MultiEntityRequest<
+  MultiEntityRequestParams,
+  EntityResponse[],
+  RequestBody,
+  MultiEntityRequestQuery & { includeRootEntity?: boolean }
+>;
+export class MultiEntityDescendantsRoute extends Route<MultiEntityDescendantsRequest> {
+  async buildResponse() {
+    const { hierarchyId, entities, fields, field, filter } = this.req.ctx;
+    const { includeRootEntity = false } = this.req.query;
+    const responseEntities: EntityType[] = [];
+    await Promise.all(
+      entities.map(async entity => {
+        const descendants = await entity.getDescendants(hierarchyId, {
+          ...filter,
+        });
+        const entitiesToUse = includeRootEntity ? [entity].concat(descendants) : descendants;
+        responseEntities.push(...entitiesToUse);
+      }),
+    );
+    if (field) {
+      return formatEntitiesForResponse(this.req.models, this.req.ctx, responseEntities, field);
+    }
+    return formatEntitiesForResponse(this.req.models, this.req.ctx, responseEntities, fields);
```

Comment:
```suggestion

  return formatEntitiesForResponse(this.req.models, this.req.ctx, responseEntities, field || fields);
```

Reply:
I'll need to add another function overload to support this, but I think it's worth it, will do 👍

Reply:
How come? Since you already call this method with one or the other, based on the `if` predicate

Reply:
If you're using function overloads, typescript will only allow you to call a function with one of the explicitly defined overloads. I do kind of like this behaviour tbh, even if it seems a bit weird, since otherwise it's hard to keep your overloads explicit, eg.
```
function combine(nums: number[]): number;
function combine(words: string[]): string;
function combine(values: number[] | string[]) {
  if (isNumber(values[0]) {
    return sum(values);
  }
  
  return values.join(' ');
}

const numberCombiner = (nums: number[]) => combine(nums); // valid
const wordCombiner = (words: string[]) => combine(words); // valid
const valueCombiner = (values: number[] | string[]) => combine(values); // Error: The call would have succeeded against this implementation, but implementation signatures of overloads are not externally visible.
```
But if you add:
```

Reply:
Interesting, thanks!